lib/axiom/fs/base/file_system.js

```diff
@@ -205,11 +205,16 @@ FileSystem.prototype.readFile = function(
   var whence = opt_whence || SeekWhence.Begin;
 
   return this.createOpenContext(path, openMode).then(
-      function(cx) {
-        return cx.open().then(function() {
-          return cx.read(offset, whence, dataType);
-        });
-  });
+    function(cx) {
+      return OpenContext.scope(cx, function() {
+        return cx.open().then(
+          function() {
```

Comment:
By "both" you mean "the whole dozen"? ;) It would certainly be nice to standardize on a single style, but I wonder if that's possible at all. What would look logical and elegant in one situation, might look clumsy and overwrought in others. Let's brainstorm, though.